### ISSUE TITLE
Multi Genre Support

### ISSUE THREAD

**Issue by foss-** [OxygenCobalt/Auxio#201]
**Currently**
Using `Abstract, Experimental, Glitch, IDM` in the `genre` field to tag some mp3 files results in Auxio detecting the genre as `Abstract, Experimental, Glitch, IDM` which results in very odd genre listings and makes tagging files very hard, since users are limited to only using a single genre, which makes finding some music very difficult and most music doesn't fit a single genre if you want to be a more precise than Pop, Rock, Jazz or Classical, which is overly broad and not very helpful.

#### Do other music players handle this? If so, how?
**Jellyfin...**
shows 4 Genre Entires, separated by comma. The great thing here is, that Jellyfin does not read this as single Genre `Abstract, Experimental, Glitch, IDM` but lists 4 separate genres `Abstract`, `Experimental`, `Glitch`, `IDM`. Each entry can be clicked to show all albums of respective genre.
![Jellyfin](https://user-images.githubusercontent.com/5708172/180832013-4adb9af9-122a-4170-b146-ff97ac1eb2c1.png)

**Plex...**
does some automatic metadata magic but behaves very similar and again each individual style is clickable to 
![plex](https://user-images.githubusercontent.com/5708172/180834454-9d187e9d-d02d-42fa-b7a2-6beee1466ff4.png)

#### Why do you think this will improve everyone's usage of Auxio?
Allowing a more flexible and fine grained genre tagging should be a win not a loss.

#### Due Diligence:
- [x] I have read the [Contribution Guidelines](https://github.com/OxygenCobalt/Auxio/blob/dev/.github/CONTRIBUTING.md).
- [x] I have read the [Accepted Additions and Requests](https://github.com/OxygenCobalt/Auxio/blob/dev/info/ADDITIONS.md) document.
- [x] I have checked for this feature in the [FAQ](https://github.com/OxygenCobalt/Auxio/blob/dev/info/FAQ.md).

**Comment by OxygenCobalt** [comment]
Sure, why not. It doesn't have the same problem that parsing for separators does in other contexts (like artists), so this is a reasonable thing to add alongside #195.

**Comment by OxygenCobalt** [comment]
Quick question @foss-. In Auxio, there are configuration options on how a song should be played when it's selected from the library or from search. One of these is "Play from genre". However, if there are several genres linked to that song, it's unclear what genre the app should play from. Is there any preference for what the app should do here? I really don't want to just drop that option when I add multi-genre support.

**Comment by foss-** [comment]
I am unfamiliar with that option. Did a search, scrolled to song results. Long pressed a song but did not find an option `Play from genre`.

Same then long pressing a song in the songs tab. I feel like I might be missing something or misunderstanding what the feature does and where to find it. Could you elaborate?

Sounds as if it would play shuffled songs from the same genre the selected song has? If so, this is a somewhat hard to solve problem since just using the first genre (assuming it is somewhat the main genre or the one the song fits with the most) can be off, when a user just uses alphabetic order for their multi-genre tagging. 🤔

**Comment by OxygenCobalt** [comment]
Go to settings -> When playing from library and set it to "Play from genre".

Currently, my idea for how to handle it is to play from the most prominent genre (i.e the one that has the most songs in general). However, I'm not sure how that would go down.

**Comment by foss-** [comment]
Activated that option. So from what I understand this will shuffle other songs from the same genre as the selected song. I thin as a user, I would need some explanatory text in settings to understand waht this feature does. But it works and I think I like it.

Your approach how to solve this for multiple genre songs is good I think, since more exotic genres (which users may still want to cover using genre tag) may lead to bad shuffle results due to the limited amount of songs in that genre. And users can always navigate to exotic genre via UI to explicitly shuffle the less known genre if they want.

**Comment by OxygenCobalt** [comment]
Okay, I think that will be my plan then for this and #195.

**Comment by illdeletethis** [comment]
The way this could be useful for me is if there were a clear genre hierarchy, meaning either last or first is the most specific, and the other end the most general.
In that case i"d prefer shuffle from the specific.
Also a setting on the genre tab for how specific the displayed categories should be would be advantageous then.
But i"m not sure i"d use it much, since retagging the whole library takes quite some time, so the feedback of foss- certainly matters more on this.

**Comment by OxygenCobalt** [comment]
Mirroring this from #195, but I dropped that idea of picking the genre with the most songs. In any situation where it is unclear what to do since there are multiple genre instances, a dialog will be shown asking the user to choose which genre to pick.

**Comment by OxygenCobalt** [comment]
Just to let you know, for consistency with #195, Auxio will likely approach multi-value tags in a way not dependent on separators, at least by default. So:

- In Vorbis, multiple instances of `GENRE` will be the standard way of detecting multiple genres.
- In ID3v2, Auxio will accept multiple null-terminated values in a single ID3v2 frame. For example, "Future Garage[NUL]House[NUL]Electronica[NUL](32)". This is ID3v2.4 convention technically, but nobody cares and these tags can be written to ID3v2.3 too.

I'll also be adding an option to interpret separators in these tags, so `, & + ;`, alongside parsing the strange legacy genre format where integer values can be chained together, like `(32)(64)Future Garage`.

**Comment by OxygenCobalt** [comment]
Multi-genre support is effectively in Auxio now. A picker dialog has been fully implemented, multi-value tags are supported, separators are supported, etc. However, it's very unlikely that the "Play from genre" setting that I spoke about prior will actually stay in the app.

Most of the UI framework around multi-genre support was made so I could easily implement it for multi-artist support. Moreover, the more I worked on this, the more I began to saw genres as closer to playlists than artists or albums. It doesn't really make sense to play from a genre, and I doubt many use it. It would definitely be easier for me to remove the setting outright and maybe re-add it if there is actually demand.

**Comment by foss-** [comment]
First off: thanks for adding multi-genre support. That is much appreciated and looking forward to it.

`Play from genre` would always be possible by navigating to genre > specific genre and and shuffle the tracklist from there, right?

**Comment by OxygenCobalt** [comment]
Yes, that should be possible @foss-. I'm just removing the option from the "When a song is played from the library" and "When a song is played from item details" settings, and only re-adding them if there is actually demand for it.

**Comment by mddvul22** [comment]
This is awesome.  I'm looking forward to this next release!  Multiple
